### PR TITLE
Give SQL transactions a debug name

### PR DESCRIPTION
**Pull request by tamird**
Just "sql" for now.

Easier to review commit-by-commit, first one is cleanup.

### CODE REVIEW

**Comment by petermattis** [comment]
LGTM

Shouldn't be difficult to pass a debug name from the client and/or allow `SET TRANSACTION NAME = foo`.

File: sql/server.go [hunk 1]
```diff
@@ -218,8 +219,9 @@ func (s *Server) execStmt(stmt parser.Statement, req driver.Request, planMaker *
 			// Start a transaction here and not in planMaker to prevent begin
 			// transaction from being called within an auto-transaction below.
 			planMaker.txn = client.NewTxn(*s.db)
+			planMaker.txn.SetDebugName("sql", 0)
```
Comment (spencerkimball) [inline sql/server.go]: Could we set this to the explained plan instead? That'd be a lot more useful.

Comment (tamird) [inline sql/server.go]: What plan? This is getting set at the moment of `BEGIN` coming in, which doesn't really do any work.

**Comment by vivekmenezes** [comment]
LGTM

**Pull request merged**